### DISCUSSION TITLE
Create Prometheus Adapter API Service

### DISCUSSION THREAD

**Discussion by gregoriomartin** [Azure/prometheus-collector#522] [Q&A] [answered]
Hi everyone,

Is it possible to create a Prometheus Adapter API Service to allow Horizontal Pod Autoscaler to read the http_requests_total metric?

Also, while navigating through Grafana metrics, I couldn't find the http_request_total metric. Should I make some adjustments to enable it?

**Comment by vishiy** [comment, marked as answer]
1) KEDA has support for horizontal scaling thru external/azure managed prometheus metrics. This support has been added to KEDA, starting with KEDA version 2.10.0. Please try it out and let us know. Its better than native HPA adapters as it can scale down to 0 (where as native HPA can do only 1, and also native HPA supports only cpu & memory based scaling by default). -- https://keda.sh/docs/2.10/scalers/prometheus/

2) Can you explain more on `http_request_total` metric? Is it the API server requests you are looking or is it any other custom/workload metric you want to collect ? If later, you can collect any custom metric thru custom scrape jobs.

**Reply by gregoriomartin** [reply]
According to the KEDA documentation we can set the scaling based on the metric `http_requests_total`. However, this metric doesn't seems to exist when I try to query it. 

![image](https://github.com/Azure/prometheus-collector/assets/26978598/64af2d9a-10df-460c-85b6-476d127bc78a)

**Comment by vishiy** [comment]
Hi - @gregoriomartin - apologies for late response. We don't collect all metrics by default. But you could customize to collect/scrape anything more/less than we collect by default. Please see here on what we collect/scrape by default and also how you can customize that behavior -- https://learn.microsoft.com/en-us/azure/azure-monitor/containers/prometheus-metrics-scrape-configuration-minimal
